classes/Http/Controller/Admin/SpeakersController.php

```diff
@@ -157,7 +159,17 @@ public function deleteAction(Request $req)
 
         $mapper = $spot->mapper(\OpenCFP\Domain\Entity\User::class);
         $speaker = $mapper->get($req->get('id'));
-        $response = $mapper->delete($speaker);
+
+        $spot->config()->connection()->beginTransaction();
+
+        try {
+            $this->removeSpeakerTalks($speaker);
+            $response = $mapper->delete($speaker);
+        } catch (\Exception $e) {
+            $spot->config()->connection()->rollBack();
+        }
+
+        $spot->config()->connection()->commit();
```

Comment:
The `commit()` call should go inside the `try` block. Otherwise, it will try to commit after a rollback is performed.